src/components/Facility/LocationManagement.tsx

```diff
@@ -42,6 +45,18 @@ export default function LocationManagement({ facilityId }: Props) {
     name: "",
     id: "",
   });
+  const [facilityMiddleware, setFacilityMiddleware] = useState<
+    string | undefined
+  >(undefined);
+
+  useTanStackQueryInstead(routes.getPermittedFacility, {
+    pathParams: { id: facilityId },
+    onResponse: (res) => {
+      if (res.data) {
+        setFacilityMiddleware(res.data?.middleware_address);
+      }
+    },
+  });
```

Comment:
this is deprecated. use new methods

https://docs.ohc.network/docs/care/development/data-fetching-in-care/